streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -338,12 +335,14 @@ private void commitState() {
         final long startNs = time.nanoseconds();
 
         stateMgr.flush();
-
         recordCollector.flush();
 
-        // we need to preserve the original partitions times before calling commit
-        // because all partition times are reset to -1 during close
+        commitSensor.record(time.nanoseconds() - startNs);
```

Comment:
Frankly, not sure if this is correct any longer. What do we want to record with this sensor exactly? Flushing can be expensive and we might want to record it as part of committing -- but I am not sure.